src/routes/userapp/components/core/CustomNumberInput.tsx

```diff
@@ -0,0 +1,192 @@
+import * as React from "react";
+import {
+  Unstable_NumberInput as NumberInput,
+  NumberInputProps as BaseNumberInputProps,
+} from "@mui/base/Unstable_NumberInput";
+import { styled } from "@mui/system";
+import RemoveIcon from "@mui/icons-material/Remove";
+import AddIcon from "@mui/icons-material/Add";
+
+interface NumberInputProps extends BaseNumberInputProps {
+  disabled: boolean;
+}
+
+const CustomNumberInput = React.forwardRef(function CustomNumberInput(
+  props: NumberInputProps,
+  ref: React.ForwardedRef<HTMLDivElement>,
+) {
+  return (
+    <NumberInput
+      slots={{
+        root: StyledInputRoot,
+        input: StyledInput,
+        incrementButton: StyledButton,
+        decrementButton: StyledButton,
+      }}
+      slotProps={{
+        incrementButton: {
+          children: <AddIcon />,
+          className: "increment",
+          disabled: props.disabled,
+        },
+        decrementButton: {
+          children: <RemoveIcon />,
+          disabled: props.disabled,
+        },
+      }}
+      {...props}
+      ref={ref}
+    />
+  );
+});
+
+interface QuantityInputProps {
+  onUserCountChange: (value: number) => void;
+  value: number;
+  disabled: boolean;
+}
+
+export default function QuantityInput({
```

Comment:
Staraj się trzymać tego żeby plik .tsx nazywał się tak samo, jak komponent który jest z niego eksportowany. Bo dla kogoś, kto tego nie pisał, to jest mocno mylące